src/persist-types/src/arrow.rs

```diff
@@ -301,6 +302,11 @@ impl ArrayOrd {
 
 /// A struct representing a particular entry in a particular array. Most useful for its `Ord`
 /// implementation, which can compare entire rows across similarly-typed arrays.
+///
+/// It is an error to compare indices from arrays with different types, with one exception:
```

Comment:
Might be nice to have a couple unit tests explicitly cover this case and also two structs where neither is a prefix of the other.

Reply:
Done!